sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
I'm using Visual Studio Community 2015 and it compiles fine here. Since both substr method and operator+ for std::string and char* were available before C++11 (which VS 2010 doesn't fully support), I suppose that maybe headers shipped with VS 2015 had their contents changed.

Try including <string> header in LoadGameHook.cpp and see if it helps.

Also, I used std::unordered_map and possibly other parts of C++11 available in VS 2010, as there's no instruction to not do so. Should I not use these in Sfall?